samcli/commands/package/artifact_exporter.py

```diff
@@ -0,0 +1,611 @@
+"""
+Logic for uploading to S3 per Cloudformation Specific Resource
+"""
+# pylint: disable=no-member
+
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import logging
+import os
+import tempfile
+import zipfile
+import contextlib
+from contextlib import contextmanager
+import uuid
+from urllib.parse import urlparse, parse_qs
+import shutil
+from botocore.utils import set_value_from_jmespath
+
+
+from samcli.commands.package import exceptions
+from samcli.yamlhelper import yaml_dump, yaml_parse
+import jmespath
+
+
+LOG = logging.getLogger(__name__)
+
+
+def is_path_value_valid(path):
+    return isinstance(path, str)
+
+
+def make_abs_path(directory, path):
+    if is_path_value_valid(path) and not os.path.isabs(path):
+        return os.path.normpath(os.path.join(directory, path))
+    return path
+
+
+def is_s3_url(url):
+    try:
+        parse_s3_url(url)
+        return True
+    except ValueError:
+        return False
+
+
+def is_local_folder(path):
+    return is_path_value_valid(path) and os.path.isdir(path)
+
+
+def is_local_file(path):
+    return is_path_value_valid(path) and os.path.isfile(path)
+
+
+def is_zip_file(path):
+    return is_path_value_valid(path) and zipfile.is_zipfile(path)
+
+
+def parse_s3_url(url, bucket_name_property="Bucket", object_key_property="Key", version_property=None):
+
+    if isinstance(url, str) and url.startswith("s3://"):
+
+        # Python < 2.7.10 don't parse query parameters from URI with custom
```

Comment:
Is there needed anymore since we have stopped support for 2.7

Reply:
good catch, its from the port, will remove this.